### ISSUE TITLE
How to  solve the problem: Unexpected label ***

### ISSUE THREAD

**Issue by zhangrui0828** [martinkersner/train-DeepLab#31]
Hi Martin, I found your tutorial very help and I am following it to train a model step by step. 
The DeepLab model : DeepLab_LargeFOV.
Dataset: original PASCAL VOC12.
Init model is vgg16_20M. 
Solver: exper/voc12/config/DeepLab_LargeFOV/solver_train.prototxt
train_net: exper/voc12/config/DeepLab_LargeFOV/train_train.prototxt
source: exper/voc12/list/train.txt

When I run run_pascal_strong.sh, an error always occur and I can't further proceed. The error is shown below.
…...
I0612 15:44:39.689864 12910 net.cpp:208] This network produces output accuracy
I0612 15:44:39.689891 12910 net.cpp:467] Collecting Learning Rate and Weight Decay.
I0612 15:44:39.689899 12910 net.cpp:219] Network initialization done.
I0612 15:44:39.689903 12910 net.cpp:220] Memory required for data: 3663953152
I0612 15:44:39.689975 12910 solver.cpp:41] Solver scaffolding done.
I0612 15:44:39.689981 12910 solver.cpp:160] Solving DeepLab-LargeFOV
I0612 15:44:39.689985 12910 solver.cpp:161] Learning Rate Policy: step
F0612 15:44:41.198884 12910 softmax_loss_layer.cpp:86] Unexpected label 19
*** Check failure stack trace: ***
    @     0x7efefd561daa  (unknown)
    @     0x7efefd561ce4  (unknown)
    @     0x7efefd5616e6  (unknown)
    @     0x7efefd564687  (unknown)
    @           0x5482a8  caffe::SoftmaxWithLossLayer<>::Forward_cpu()
    @           0x47413a  caffe::Net<>::ForwardFromTo()
    @           0x4743cf  caffe::Net<>::ForwardPrefilled()
    @           0x57a320  caffe::Solver<>::Solve()
    @           0x419be8  train()
    @           0x412278  main
    @     0x7efefa399f45  (unknown)
    @           0x417927  (unknown)
    @              (nil)  (unknown)
Aborted

The number after “Unpexted label” is always different at every time. 
Do you have any ideas on this? That would be really greatful.
Thanks!

**Comment by debayan** [comment]
Your image needs to be single channel and it should not have a larger number in the pixel value than the number of classes you have specified in your softmax layer.
